pkg/generate/app/cmd/newapp.go

```diff
@@ -777,7 +777,9 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	if len(components.UseSource()) > 1 && len(c.To) > 0 {
 		return nil, errors.New("only one component with source can be used when specifying an output image reference")
 	}
-
+	if len(c.ContextDir) > 0 && len(c.TemplateFiles) > 0 {
+		return nil, errors.New("--context-dir is not supported when using a template")
+	}
```

Comment:
just to confirm, c.TemplateFiles is populated for all of the following conditions:

1) "oc new-app mylocaltemplate.yaml"
2) "oc new-app -f mylocaltemplate.yaml"
3) "oc new-app http://myremote/template.yaml"
4) "oc new-app -f http://myremote/template.yaml"
5) "oc new-app myclustertemplate"

?